# Accelerating Cleantech Advancements through NLP-Powered Text Mining and Knowledge Extraction

***Team: Rebecca Gergess, Tejesh Reddy Koki, Sasa Ljubisavlijevic***

## Stage 1: Data Cleaning and Preprocessing and EDA

In [34]:
#Loading Packages
import pandas as pd
import re
import string

Data Collection and Cleaning.

o Acquire a comprehensive dataset of cleantech texts.

o Perform data cleaning to remove irrelevant information, special characters, and duplicates.

In [45]:
# Setting the path and Loading the data from csv file
from pathlib import Path
print(Path.cwd())

data_folder = Path("dataset/cleantech_media_dataset_v1_20231109.csv")
data_folder


c:\Users\kokit\OneDrive - Hochschule Luzern\Study Docuemnts\Semester-4\CLT\Project\NLP


WindowsPath('dataset/cleantech_media_dataset_v1_20231109.csv')

In [46]:
#Opening the kaggle dt file and assigning it to a variable
data = pd.read_csv(data_folder)
data

Unnamed: 0                                              title  \
0           1280  Qatar to Slash Emissions as LNG Expansion Adva...   
1           1281               India Launches Its First 700 MW PHWR   
2           1283              New Chapter for US-China Energy Trade   
3           1284  Japan: Slow Restarts Cast Doubt on 2030 Energy...   
4           1285     NYC Pension Funds to Divest Fossil Fuel Shares   
...          ...                                                ...   
9602       82339  Strata Clean Energy Nets $ 300 Million in Fund...   
9603       82340  Orsted Deploying SparkCognition Renewable Suit...   
9604       82341     Veolia Has Plans for 5 MW of Solar in Arkansas   
9605       82342                      SunEdison: Too Big, Too Fast?   
9606       82343  Vikings Solar-Plus-Storage Development Nets Fi...   

            date author                                            content  \
0     2021-01-13    NaN  ["Qatar Petroleum ( QP) is targeting aggressiv...   
1     2021-01-15    NaN  ["• Nuclear Power Corp. of India Ltd. ( NPCIL)...   
2     2021-01-20    NaN  ["New US President Joe Biden took office this ...   
3     2021-01-22    NaN  ["The slow pace of Japanese reactor restarts c...   
4     2021-01-25    NaN  ["Two of New York City's largest pension funds...   
...          ...    ...                                                ...   
9602  2023-11-06    NaN  ['Strata Clean Energy has closed a $ 300 milli...   
9603  2023-11-07    NaN  ['Global renewable energy developer Ørsted is ...   
9604  2023-11-07    NaN  ['Veolia North America, a provider of environm...   
9605  2023-11-08    NaN  ['Once the self-proclaimed “ leading renewable...   
9606  2023-11-08    NaN  ['Arevon Energy Inc. has closed financing on t...   

                domain                                                url  
0          energyintel  https://www.energyintel.com/0000017b-a7dc-de4c...  
1          energyintel  https://www.energyintel.com/0000017b-a7dc-de4c...  
2          energyintel  https://www.energyintel.com/0000017b-a7dc-de4c...  
3          energyintel  https://www.energyintel.com/0000017b-a7dc-de4c...  
4          energyintel  https://www.energyintel.com/0000017b-a7dc-de4c...  
...                ...                                                ...  
9602  solarindustrymag  https://solarindustrymag.com/strata-clean-ener...  
9603  solarindustrymag  https://solarindustrymag.com/orsted-deploying-...  
9604  solarindustrymag  https://solarindustrymag.com/veolia-has-plans-...  
9605  solarindustrymag  http://www.solarindustrymag.com/online/issues/...  
9606  solarindustrymag  https://solarindustrymag.com/vikings-solar-plu...  

[9607 rows x 7 columns]

We can see that the first column has no heading and the number seems unique for each row of data. We can assume this as its id. Secondly, in the column content we can see there are square brackets and double inverted commas, so cleaning would be required here.

In [47]:
data.columns.values[0] = 'ID'
print(data[:2])

     ID                                              title        date author  \
0  1280  Qatar to Slash Emissions as LNG Expansion Adva...  2021-01-13    NaN   
1  1281               India Launches Its First 700 MW PHWR  2021-01-15    NaN   

                                             content       domain  \
0  ["Qatar Petroleum ( QP) is targeting aggressiv...  energyintel   
1  ["• Nuclear Power Corp. of India Ltd. ( NPCIL)...  energyintel   

                                                 url  
0  https://www.energyintel.com/0000017b-a7dc-de4c...  
1  https://www.energyintel.com/0000017b-a7dc-de4c...  


In [48]:
# Remove the [] and "" from the content field
def remove_punctuation(text):
    regex = re.compile ('[%s]' %re.escape(string.punctuation))
    return regex.sub('', str(text))

data['content'] = data['content'].apply(remove_punctuation)

print(data[:2])

     ID                                              title        date author  \
0  1280  Qatar to Slash Emissions as LNG Expansion Adva...  2021-01-13    NaN   
1  1281               India Launches Its First 700 MW PHWR  2021-01-15    NaN   

                                             content       domain  \
0  Qatar Petroleum  QP is targeting aggressive cu...  energyintel   
1  • Nuclear Power Corp of India Ltd  NPCIL synch...  energyintel   

                                                 url  
0  https://www.energyintel.com/0000017b-a7dc-de4c...  
1  https://www.energyintel.com/0000017b-a7dc-de4c...  


In [49]:
#checking if the author column has any other data expect NaN

has_non_nan_values = pd.notna(data['author']).any()
nan_count = data['author'].isna().sum()
print(nan_count)

# Print the result
if has_non_nan_values:
    print("\nThe 'author' column has values other than NaN.")
else:
    print("\nThe 'author' column only contains NaN values.")

9576

The 'author' column has values other than NaN.


In total of 9607 rows, 9676 rows contain no author name. So author is not impacting much of our results if we drop and any way the main topic for our case analysis is Content. Keeping this in mind, dropping url, author.

In [50]:
data['content'][0]

'Qatar Petroleum  QP is targeting aggressive cuts in its greenhouse gas emissions as it prepares to launch Phase 2 of its planned 48 million ton per year LNG expansion In its latest Sustainability Report published on Wednesday QP said its goals include  reducing the emissions intensity of Qatars LNG facilities by 25 and of its upstream facilities by at least 15  The company is also aiming to reduce gas flaring intensity across its upstream facilities by more than 75 and has raised its carbon capture and storage ambitions from 5 million tonsyr to 7 million tonsyr by 2027 About 22 million tonsyr of the carbon capture goal will come from the 32 million tonyr Phase 1 of the LNG expansion also known as the North Field East project A further 11 million tonsyr will come from Phase 2 known as the North Field South project which will raise Qatars LNG capacity by a further 16 million tonsyr Qatar currently has an LNG production capacity of around 78 million tonsyr and is eyeing a phased expansio

      ID                                              title        date  \
0    NaN  Qatar to Slash Emissions as LNG Expansion Adva...  2021-01-13   
1    NaN               India Launches Its First 700 MW PHWR  2021-01-15   
2    NaN              New Chapter for US-China Energy Trade  2021-01-20   
3    NaN  Japan: Slow Restarts Cast Doubt on 2030 Energy...  2021-01-22   
4    NaN     NYC Pension Funds to Divest Fossil Fuel Shares  2021-01-25   
...   ..                                                ...         ...   
9602 NaN  Strata Clean Energy Nets $ 300 Million in Fund...  2023-11-06   
9603 NaN  Orsted Deploying SparkCognition Renewable Suit...  2023-11-07   
9604 NaN     Veolia Has Plans for 5 MW of Solar in Arkansas  2023-11-07   
9605 NaN                      SunEdison: Too Big, Too Fast?  2023-11-08   
9606 NaN  Vikings Solar-Plus-Storage Development Nets Fi...  2023-11-08   

                                                content            domain  
0     Qatar Petroleum  